ui/lib/kv/addon/components/page/secret/edit.hbs

```diff
@@ -9,11 +9,61 @@
   </p.levelLeft>
 </PageHeader>
 
-<KvSecretForm
-  @previousVersion={{@previousVersion}}
-  @metadata={{@metadata}}
-  @secret={{@newVersion}}
-  {{! onSave transition to parent secret route to re-query latest version }}
-  @onSave={{transition-to "vault.cluster.secrets.backend.kv.secret"}}
-  @onCancel={{transition-to "vault.cluster.secrets.backend.kv.secret.details"}}
-/>
\ No newline at end of file
+<Toolbar>
+  <ToolbarFilters>
+    <Toggle @name="json" @status="success" @size="small" @checked={{this.showJsonView}} @onChange={{this.toggleJsonView}}>
+      <span class="has-text-grey">JSON</span>
+    </Toggle>
+  </ToolbarFilters>
+</Toolbar>
```

Comment:
KvPageHeader can be passed in filters and it should style this all for you. However, maybe it's because you need control of the onChange?

Reply:
Cool - that makes sense. And yeah - I think once all of the views are rounded out we can revisit and maybe refactor the header component so that it can meaningfully be used in each view!